provider/core/src/key.rs

```diff
@@ -141,7 +141,7 @@ impl DataKeyPath {
     #[inline]
     pub const fn get(self) -> &'static str {
         /// core::slice::from_raw_parts(a, b) = core::mem::transmute((a, b)) hack
-        /// ```compile_fail
+        /// ```
```

Comment:
Please remove this test and update the code below to use `core::slice::from_raw_parts` instead of `core::mem::transmute`.